src/monodroid/jni/basic-android-system.cc

```diff
@@ -12,13 +13,29 @@ size_t BasicAndroidSystem::app_lib_directories_size = 0;
 const char* BasicAndroidSystem::built_for_abi_name = nullptr;
 
 void
-BasicAndroidSystem::setup_app_library_directories (jstring_array_wrapper& runtimeApks, jstring_array_wrapper& appDirs, int androidApiLevel)
+BasicAndroidSystem::detect_embedded_dso_mode (jstring_array_wrapper& appDirs) noexcept
 {
-	if (androidApiLevel < 23 || !is_embedded_dso_mode_enabled ()) {
+	// appDirs[2] points to the native library directory
+	simple_pointer_guard<char[]> libmonodroid_path = utils.path_combine (appDirs[2].get_cstr (), "libmonodroid.so");
```

Comment:
Should this verify that `appDirs[2]` is a valid index before using it?  Or will that  cause an immediate exception/SIGSEGV if `appDirs[2]` is invalid?

Reply:
Won't cause an exception; instead `appDirs[2]` will return an "invalid string", and `.get_cstr()` will return `nullptr`.

Reply:
Result is that if `appDris[2]` isn't valid then `libmonodroid_path` will be `"libmonodroid.so"` (no directory info), which will thus depend upon "current-working-directory" behavior within **stat**(2).

Reply:
It's tied tightly with our Java code - if we don't send enough indices, let it segfault, we'll notice it and fix.